legacy-fabric-keybindings-api-v1/1.6.4/src/main/java/net/legacyfabric/fabric/mixin/client/keybinding/SettingsScreenMixin.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2020 - 2024 Legacy Fabric
+ * Copyright (c) 2016 - 2022 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.legacyfabric.fabric.mixin.client.keybinding;
+
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.screen.SettingsScreen;
+import net.minecraft.client.gui.widget.ButtonWidget;
+import net.minecraft.client.option.GameOptions;
+
+import net.legacyfabric.fabric.impl.client.keybinding.DynamicControlsOptionsScreen;
+
+@Mixin(SettingsScreen.class)
+public class SettingsScreenMixin extends Screen {
+	@Shadow
+	@Final
+	private GameOptions options;
+
+	@Inject(method = "buttonClicked", at = @At("HEAD"), cancellable = true)
+	private void hijackControlScreenButton(ButtonWidget buttonWidget, CallbackInfo ci) {
```

Comment:
This is very concerning. Why can’t we inject into the existing screen? And how would other mods be able to inject into it if we hijack it?

Reply:
I guess we can inject in vanilla control screen constructor instead?